tls/s2n_server_extensions.c

```diff
@@ -74,7 +76,7 @@ int s2n_server_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *
      * is equivalent to allowing only the uncompressed point format. Let's send the
      * extension in case clients(Openssl 1.0.0) don't honor the implied behavior.
      */
-    if (conn->secure.cipher_suite->key_exchange_alg->flags & S2N_KEY_EXCHANGE_ECC)  {
+    if (s2n_server_can_send_ec_point_formats(conn))  {
```

Comment:
This bug was found because we disable extensions for SSLv3 in our GnuTLS test: https://github.com/awslabs/s2n/blob/master/tests/integration/s2n_handshake_test_gnutls.py#L173-L174